### PR TITLE
Remove codemirror-rails

### PR DESCRIPTION
**Pull request by lvillen**
**What this PR does / why we need it**

Rails 6 does not support codemirror-rails gem, as mentioned in this [issue](https://github.com/fixlr/codemirror-rails/issues/61), hence this PR intends to remove it and add codemirror package.  

**With codemirror as a package:** 
Captura de pantalla 2023-04-20 a las 12 02 05

Captura de pantalla 2023-04-20 a las 12 02 49

**Currently at staging:** 
Captura de pantalla 2023-04-20 a las 12 03 10

Captura de pantalla 2023-04-20 a las 12 02 59

**Special notes for your reviewer**:

### CODE REVIEW

**Review by akostadinov — DISMISSED** [review]
Such change is very welcome!

**Comment by josemigallas** [comment]
> > Out of curiosity, how big is the new pack `codemirror`?
> 
> ```
>   package            size       minified   gzipped
>   codemirror@5.65.9  393.91 KB  169.58 KB  55.8 KB
> ```

Not the package but the pack.

File: app/javascript/packs/codemirror.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import CodeMirror from 'codemirror'
+import 'codemirror/mode/htmlmixed/htmlmixed'
+import 'codemirror/lib/codemirror.css'
+import 'codemirror/theme/neat.css'
+
+import './codemirror-cms.scss'
```
Comment (josemigallas) [inline app/javascript/packs/codemirror.ts:6]: Remove this import and either rename the pack to `codemirror-cms` OR the scss to `codemirror`.

File: app/javascript/packs/codemirror.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import CodeMirror from 'codemirror'
+import 'codemirror/lib/codemirror.css'
+
+window.CodeMirror = CodeMirror
```
[the remark this answers is not in the document]
Comment (josemigallas) [reply app/javascript/packs/codemirror.ts:8]: @lvillen

[the remark this answers is not in the document]
Comment (josemigallas) [reply app/javascript/packs/codemirror.ts:8]: Either use window.Codemirror or import via webpack, bot not both.

File: app/javascript/src/ActiveDocs/useCodeMirror.ts [hunk 3]
```diff
@@ -1,12 +1,18 @@
+import 'codemirror/lib/codemirror.css'
+import 'codemirror/theme/neat.css'
+import 'codemirror/mode/javascript/javascript'
+import CodeMirror from 'codemirror'
 import { useEffect } from 'react'
 
 import type { Editor } from 'codemirror'
 
+import '../../packs/codemirror-cms.scss'
```
Comment (josemigallas) [inline app/javascript/src/ActiveDocs/useCodeMirror.ts:9]: This should be done at the pack's level
```suggestion
```

File: app/views/provider/admin/cms/_includes.html.erb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <%= javascript_include_tag 'provider/cms' %>
-<%= javascript_include_tag 'provider/codemirror' %>
+<%= javascript_pack_tag 'codemirror' %>
```
Comment (josemigallas) [inline app/views/provider/admin/cms/_includes.html.erb:2]: I don't like the idea of this "incudes" html. Include the pack at the uppermost possible level.

File: app/javascript/packs/codemirror-cms.scss [hunk 5]
```diff
@@ -1,6 +1,59 @@
...
+$codemirror-bg: $content-background;
+$highlight-color: $lochMaraBlue;
+$border-color: $altoGray;
+$select-color: $border-color;
+$success-color: $appleGreen;
+$link-color: $highlight-color;
+$warning-color: $tahitiGold;
+$error-color: $guardsmanRed;
+
+// Comes from _theme.scss
```
Comment (josemigallas) [inline app/javascript/packs/codemirror-cms.scss:23]: Does it though? I don't see this in theme.scss

File: app/javascript/packs/codemirror-cms.scss [hunk 6]
```diff
@@ -1,6 +1,59 @@
-@import 'codemirror';
-@import 'codemirror/liquid';
-@import 'codemirror/themes/neat';
+// Comes from _colors.scss
```
Comment (josemigallas) [inline app/javascript/packs/codemirror-cms.scss:1]: Let's be specific
```suggestion
// Copied from _colors.scss
```

**Review by josemigallas — CHANGES_REQUESTED** [review]

File: app/views/admin/api_docs/base/_form.html.slim [hunk 7]
```diff
@@ -1,5 +1,5 @@
-- content_for :javascripts do
-  = javascript_include_tag 'provider/codemirror'
+- content_for :javascripts do
```
Comment (josemigallas) [inline app/views/admin/api_docs/base/_form.html.slim:1]: Trailing space alert 🚨

**Comment by josemigallas** [comment]
**Note for reviewers:**
These are all the pages where the codemirror input is used and might be impacted:
* ./app/views/provider/admin/cms/portlets/new.html.erb:8
* ./app/views/provider/admin/cms/builtin_legal_terms/edit.html.erb:15
* ./app/views/provider/admin/cms/builtin_legal_terms/new.html.erb:10
* ./app/views/provider/admin/cms/_content_tabs.html.erb:21
* ./app/views/provider/admin/cms/_content_tabs.html.erb:25
* ./app/views/provider/admin/cms/email_templates/edit.html.erb:17
* ./app/views/provider/admin/cms/email_templates/new.html.erb:11
* ./app/views/provider/admin/cms/layouts/new.html.erb:7
* ./app/views/provider/admin/cms/pages/new.html.erb:7
* ./app/views/provider/admin/cms/partials/new.html.erb:7

**Comment by josemigallas** [comment]
Verified:

- [ ] ./app/views/provider/admin/cms/portlets/new.html.erb:8
- [ ] ./app/views/provider/admin/cms/builtin_legal_terms/edit.html.erb:15
- [ ] ./app/views/provider/admin/cms/builtin_legal_terms/new.html.erb:10
- [ ] ./app/views/provider/admin/cms/_content_tabs.html.erb:21
- [ ] ./app/views/provider/admin/cms/_content_tabs.html.erb:25
- [ ] ./app/views/provider/admin/cms/email_templates/edit.html.erb:17
- [ ] ./app/views/provider/admin/cms/email_templates/new.html.erb:11
- [ ] ./app/views/provider/admin/cms/layouts/new.html.erb:7
- [ ] ./app/views/provider/admin/cms/pages/new.html.erb:7
- [ ] ./app/views/provider/admin/cms/partials/new.html.erb:7

File: app/javascript/packs/codemirror.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import CodeMirror from 'codemirror'
+import 'codemirror/addon/mode/overlay'// Seems to solve https://github.com/3scale/porta/blob/25fa9bc53bdd3eeb8f2a9c8e882f771c85c3c3cb/app/views/provider/admin/cms/_codemirror.html.erb#L34
+import 'codemirror/mode/css/css' // Formats and styles for CSS; Used in CMS
+import 'codemirror/mode/htmlmixed/htmlmixed' // Formats and syles for HTML and Liquid; Used in CMS
+import 'codemirror/mode/javascript/javascript' // Formats and syles for JSON; Used in ActiveDocs and in CMS
+import 'codemirror/mode/xml/xml' // Formats and syles for XML
```
Comment (akostadinov) [inline app/javascript/packs/codemirror.ts:6]: Is it used for the markdown support in portal as well?

Comment (lvillen) [reply app/javascript/packs/codemirror.ts:6]: No, it isn't 🚀

**Review by josemigallas — APPROVED** [review]

**Review by akostadinov — APPROVED** [review]

**Pull request merged**